Tzkt.Api/Websocket/Processors/CyclesProcessor.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+
+using Tzkt.Api.Repositories;
+using Tzkt.Api.Services.Cache;
+
+namespace Tzkt.Api.Websocket.Processors
+{
+    public class CycleProcessor<T> : IHubProcessor where T : Hub
+    {
+        #region static
+        const string CycleGroup = "cycle";
+        const string CycleChannel = "cycle";
+        static readonly SemaphoreSlim Sema = new(1, 1);
+        static readonly Dictionary<int, HashSet<string>> DelaySubs = new();
+        #endregion
+
+        readonly StateCache StateCache;
+        readonly StateRepository StateRepo;
+        readonly IHubContext<T> Context;
+        readonly ILogger Logger;
+
+        private int cycleStartLevel = 0;
```

Comment:
Please, use pascal case for global fields/props.

Reply:
Both properties removed now.